presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisRecordSet.java

```diff
@@ -290,6 +293,23 @@ private void getKinesisRecords()
             messagesRead += kinesisRecords.size();
         }
 
+        private byte[] uncompressIfNeeded(byte[] data)
+        {
+            if (data == null || data.length < 2) {
+                return data;
+            }
+            int magic = data[0] & 0xff | ((data[1] << 8) & 0xff00);
+            if (magic == GZIPInputStream.GZIP_MAGIC) {
```

Comment:
Does this support only GZIP or ?

Reply:
How about wrapping this logic in a `RowDecoder` which decompresses the row and passes it to the underlying decoder.

Reply:
My idea was to limit it now to Kinesis, since other consumer libraries like Kafka and Redis might have an option to uncompress it.